framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2053,100 +1634,551 @@ public final String getOption(String name, String defaultValue) {
+     */
+    @Override
+    public final Set<String> getSupportedAnnotationTypes() {
+
+        SupportedAnnotationTypes supported =
+                this.getClass().getAnnotation(SupportedAnnotationTypes.class);
+        if (supported != null) {
+            throw new BugInCF(
+                    "@SupportedAnnotationTypes should not be written on any checker;"
+                            + " supported annotation types are inherited from SourceChecker.");
+        }
+        return Collections.singleton("*");
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    /// Warning suppression and unneeded warnings
+    ///
+
+    /**
+     * @return collection of lower-case string keys that a checker honors for suppressing warnings
+     *     and errors that it issues. Each such key suppresses all warnings issued by the checker.
+     * @see SuppressWarningsKeys
+     */
+    public Collection<String> getSuppressWarningsKeys() {
+        return getStandardSuppressWarningsKeys();
+    }
+
+    /**
+     * Only ever called once; the value is cached in field {@link #suppressWarnings}.
+     *
+     * @param options the command-line options
+     * @return the argument to -AsuppressWarnings, split on commas, or null if no such argument
+     */
+    private String @Nullable [] createSuppressWarnings(Map<String, String> options) {
+        if (!options.containsKey("suppressWarnings")) {
+            return null;
+        }
+
+        String swString = options.get("suppressWarnings");
+        if (swString == null) {
+            return null;
+        }
+
+        return arrayToLowerCase(swString.split(","));
+    }
+
+    /**
+     * Issues a warning about any {@code @SuppressWarnings} that isn't used by this checker, but
+     * contains a key that would suppress a warning from this checker.
+     */
+    protected void warnUnneededSuppressions() {
+        if (!hasOption("warnUnneededSuppressions")) {
+            return;
+        }
+
+        Set<Element> elementsSuppress = new HashSet<>(this.elementsWithSuppressedWarnings);
+        this.elementsWithSuppressedWarnings.clear();
+        Set<String> checkerKeys = new HashSet<>(getSuppressWarningsKeys());
+        Set<String> errorKeys = new HashSet<>(messagesProperties.stringPropertyNames());
+        warnUnneedSuppressions(elementsSuppress, checkerKeys, errorKeys);
+        getVisitor().treesWithSuppressWarnings.clear();
+    }
+
+    /**
+     * Issues a warning about any {@code @SuppressWarnings} that isn't used by this checker, but
+     * contains a key that would suppress a warning from this checker.
+     *
+     * @param elementsSuppress elements with a {@code @SuppressWarnings} that actually suppressed a
+     *     warning
+     * @param checkerKeys suppress warning keys that suppress any warning from this checker
+     * @param errorKeys error keys that can be issued by this checker
+     */
+    protected void warnUnneedSuppressions(
```

Comment:
Should this method be called `warnUnneededSuppressions`, like the other method above?
How does a developer chose between the two version?

Reply:
That was the name before, but I have changed it.